Add component yaml configuration definition with connection env renaming guide (#1684)
wso2/docs-choreo-dev: en/docs/develop-components/sharing-and-reusing/use-a-connection-in-your-service.md

          | ChoreoAPIKey            | CHOREO_<CONNECTION_NAME\>CHOREOAPIKEY                          |

>           If you'd like to use custom environment variable names instead of the Choreo-defined ones, add env configurations under `configuration` in the same file.

Reviewer: ```suggestion
          If you want to use custom environment variable names instead of the Choreo-defined ones, you must define the environment configurations under the `configuration` section in the same file.
```

          The following is a sample snippet: 

          ``` yaml
          configuration:
            env:
              - name: HR_SERVICE_URL
                valueFrom:
                  connectionRef:
                    name: hr-connection
                    key: ServiceURL
          ```

>           The following table provides details on the configuration keys associated with the connection, which can be used as `env[i].valueFrom.connectionRef.key`.

Reviewer: ```suggestion
          The following table provides details on the configuration keys associated with the connection. These keys can be referenced as `env[i].valueFrom.connectionRef.key`.
```

          | TokenURL       | string     | Token URL of the STS                  | false         | false        |
          | ChoreoAPIKey   | string     | API key of the Choreo service         | false         | true         |

>           For more details, refer to the instructions under the `component.yaml file (v1.1)` tab.

Reviewer: ```suggestion
          For more details, see the instructions under the **component.yaml file (v1.1)** tab.
```
